components/odh-notebook-controller/Dockerfile.konflux

```diff
@@ -9,32 +9,21 @@ ARG TARGETARCH
 ## Build args to be used at this step
 ARG SOURCE_CODE
 
-WORKDIR /workspace
-
+WORKDIR /opt/rhods
 # Copy the Go Modules manifests
 COPY ${SOURCE_CODE}/notebook-controller ./notebook-controller
-# This is necessary because the Jupyter controller now depends on
-# components/common
-COPY ${SOURCE_CODE}/common ./common
-
-# Copy the Go Modules manifests
-COPY notebook-controller /workspace/notebook-controller
-COPY common /workspace/common
-
-# cache deps before building and copying source so that we don't need to re-download as much
-# and so that source changes don't invalidate our downloaded layer
-RUN cd /workspace/notebook-controller && go mod download
```

Comment:
this is the usual strategy about minimizing container rebuild time when go.mod does not change; I'm not sorry to see it go, but I would not mind keeping it

we can restore it in the future ourselves, at some point; for now I would just let konflux migration proceed